usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/PetData.java

```diff
@@ -110,6 +135,106 @@ public void setRecommendedKcal(Double recommendedKcal) {
+            || field.equals(NEEDS))) && !(newValue instanceof String)) {
+            throw new IllegalArgumentException("New value must be a String");
+        } else if (field.equals(RECOMMENDED_KCAL) && !(newValue instanceof Double)) {
+            throw new IllegalArgumentException("New value must be a Double");
+        } else if (field.equals(GENDER) && !(newValue instanceof GenderType)) {
+            throw new IllegalArgumentException("New value must be a GenderType");
+        }
+    }
+
+    /**
+     * Checks that field has the correct format for a Pet collection attribute.
+     * @param field Name of the attribute collection. Possible fields: meals, trainings, washes, weights
+     */
+    public static void checkCollectionField(String field) {
+        if (!MEALS.equals(field) && !TRAININGS.equals(field) && !WASHES.equals(field)
+            && !WEIGHTS.equals(field)) {
+            throw new IllegalArgumentException("Field does not exists");
+        }
+    }
+
+    /**
+     * Checks that field, key and body have the correct format of a Pet attribute.
+     * @param field Name of the attribute collection. Possible fields: meals, trainings, washes, weights
+     * @param key Key of the attribute
+     * @param body Body of the attribute
+     */
+    public static void checkCollectionKeyAndBody(String field, String key, Map<String, Object> body) {
+        switch (field) {
+            case MEALS:
+                checkMeals(key, body);
+                break;
+            case TRAININGS:
+            case WASHES:
+            case WEIGHTS:
+                checkDateAndValueInteger(key, body);
+                break;
+            default:
+                throw new IllegalArgumentException("Field does not exists");
+        }
+    }
+
+    /**
+     * Checks that key and body have the correct format for a meal.
+     * @param key Key of the attribute
+     * @param body Body of the attribute
+     */
+    public static void checkMeals(String key, Map<String, Object> body) {
+        checkDateFormat(key);
+        if (body.size() != 2 || !body.containsKey("kcal") || !body.containsKey("mealName")) {
+            throw new IllegalArgumentException("Request body does not have a correct format");
+        }
+        if (!(body.get("kcal") instanceof Double) || !(body.get("mealName") instanceof String)) {
+            throw new IllegalArgumentException("Request body does not have a correct format");
+        }
+    }
+
+    /**
+     * Checks that key and body have the correct format for a date key and body with one element whose key is 'value'
+     * and has an Object of type Integer.
+     * @param key Key of the attribute
+     * @param body Body of the attribute
+     */
+    public static void checkDateAndValueInteger(String key, Map<String, Object> body) {
+        checkDateFormat(key);
+        if (body.size() != 1 || !body.containsKey("value")) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The String "value" appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5479466)